src/backend/marsha/core/models/video.py

```diff
@@ -298,6 +303,59 @@ def recording_time(self):
             )
         return recording_time
 
+    @property
+    def live_duration(self):
+        """
+        Return the duration since the live started.
+        """
+        end = self.live_ended_at
+        start = self.live_info.get("started_at")
+        if end and start:
+            return int(end) - int(start)
+        return 0
+
+    @property
+    def live_ended_at(self):
+        """
+        Return the timestamp at which the live ended or current timestamp if it's still running
+        """
+        if self.live_state == RUNNING:
+            return int(to_timestamp(timezone.now()))
+
+        if self.live_info.get("stopped_at"):
+            return int(self.live_info.get("stopped_at"))
+
+        return None
+
+    def get_list_timestamps_attendences(self):
+        """
+        Depending on the duration of the video, build an array of timestamps.
+        This array is cached for a duration.
+        """
+
+        # first try to get this information from the cache
+        cache_key = f"video:{self.id}:get_list_timestamps_attendences"
+        timestamp_list = cache.get(cache_key, {})
+
+        # key exists from the cache, no need to do anything
+        if timestamp_list:
+            return timestamp_list
+
+        duration = self.live_duration
+        if duration > settings.ATTENDANCE_POINTS:
+            # we expect to have a minimum a point per seconds
+            started = int(self.live_info.get("started_at"))
+            elapsed = round(duration / (settings.ATTENDANCE_POINTS - 1))
+            for sec in range(started, self.live_ended_at + 1, elapsed):
+                timestamp_list[str(sec)] = {}
+
+            # save key
+            cache.set(
```

Comment:
With the last modification you made I think you don't need cache here anymore.

Reply:
I think it's better if we keep it to give a chance for page 2 to be on the same timeline don't you think.